validator/sawtooth_validator/journal/publisher.py

```diff
@@ -496,6 +496,10 @@ def __init__(self,
         self._check_publish_block_frequency = check_publish_block_frequency
         self._publisher_thread = None
 
+        # A series of states that allow us to check for condition changes.
+        # These can be used to log only at the boundary of condition changes.
+        self._logging_states = {}
```

Comment:
I would prefer a python object with explicit attributes to a dictionary, since it is easy to abuse the dictionary

Reply:
Since this is a pattern we are likely to repeat elsewhere, it may also be worth thinking about how to do this consistently in other objects.

Reply:
Not all states will potentially be boolean, hence the choice of dict here.  How about we split the difference and go with a named tuple?